### PR TITLE
Fix OpenURL escaping

### LINKED ISSUE #11182: [BUG] Forward slash is changed to %2F in link
When inserting a link using `TapTool` and `OpenURL`, formatting with a `@link` type string, any forward slash character is converted to `%2F`, making the link broken. I found this behaviour in Bokeh 2.3.0 and 2.3.1, but not in 2.2.3, where the forward slashes are correctly included in the link. 

I'm using python 3.9.4 (from conda) on Ubuntu 20.04.2 LTS, with notebook 6.3.0, tested in both firefox 87.0 and chromium 90.0.4430.72.

Below is an MWE, adopted from the [docs](https://docs.bokeh.org/en/latest/docs/user_guide/interaction/callbacks.html#openurl)
```
from bokeh.models import ColumnDataSource, OpenURL, TapTool
from bokeh.plotting import figure, output_file, show

output_file("openurl.html")

p = figure(plot_width=400, plot_height=400,
           tools="tap", title="Click the Dots")

source = ColumnDataSource(data=dict(
    x=[1, 2],
    y=[2, 5],
    link=["en/latest/",
          "en/latest/docs/reference/models/callbacks.html?highlight=openurl#bokeh.models.callbacks.OpenURL", 
          ]
    ))

p.circle('x', 'y', size=20, source=source)

url = "https://docs.bokeh.org/@link"
taptool = p.select(type=TapTool)
taptool.callback = OpenURL(url=url)

show(p)

```

When run in bokeh 2.3.0 or 2.3.1, upon clicking the dot at (x=1, y=2) ,the browser opens https://docs.bokeh.org/en%2Flatest%2F, rather than https://docs.bokeh.org/en/latest/ which is opened when using bokeh 2.2.3.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #11182
- [x] tests added / passed

This fixes the issues. @mattpap any thoughts on testing? I briefly looked at using `sinon.spy` but some additional refactoring would be needed to make that work

### CODE REVIEW

**Comment by mattpap** [comment]
With `sinon` you should be able to spy on `window.open()` and its arguments, so I don't think any refactorings are needed (at least this point).

**Comment by bryevdv** [comment]
Ah, for sure, I was coming from the opposite angle, trying to patch `replace_placeholders`. Since this is so simple I will actually add a full set of tests tonight to check selection handling or more escaping cases.

**Comment by bryevdv** [comment]
@mattpap any idea about the BokehJS-CI failure? This PR is branched off latest `branch-2.3`

**Comment by mattpap** [comment]
It's spurious, but if persists, there may be need for adding `.allowing(N)` to that particular test.

**Comment by bryevdv** [comment]
@mattpap added `allowing` but I still can't get it to pass even after several retries

File: bokehjs/test/integration/regressions.ts [hunk 1]
```diff
@@ -730,7 +730,7 @@ describe("Bug", () => {
   })
 
   describe("in issue #10695", () => {
-    it("prevents showing MultiChoice's dropdown menu", async () => {
+    it.allowing(8)("prevents showing MultiChoice's dropdown menu", async () => {
```
Comment (mattpap) [inline bokehjs/test/integration/regressions.ts:733]: It should be 16 pixels.

File: bokehjs/src/lib/models/callbacks/open_url.ts [hunk 2]
```diff
@@ -29,16 +29,19 @@ export class OpenURL extends Callback {
     }))
   }
 
+  raise(url: string): void {
```
Comment (mattpap) [inline bokehjs/src/lib/models/callbacks/open_url.ts:32]: Maybe this is some kind of idiom I don't know, but I wouldn't associate this name with the following implementation, without seeing the implementation.

File: bokehjs/test/integration/regressions.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/test/integration/regressions.ts:733]:
```suggestion
    it.allowing(16)("prevents showing MultiChoice's dropdown menu", async () => {
```

**Comment by mattpap** [comment]
I guess one more needs to be added:
```
✗ Bug
 └┬─ in issue #10507
  └── prevents changing MultiSelect.disabled property
images differ by 22px (0.09%)
```
I will need to revisit this at some point.

File: bokehjs/src/lib/models/callbacks/open_url.ts (hunk 2 above)
Comment (bryevdv) [reply bokehjs/src/lib/models/callbacks/open_url.ts:32]: I am using it in the sense of "raise a browser window" which is common enough at least  in my experience, but I'm happy to change it if you have any specific suggestion.

File: bokehjs/test/integration/regressions.ts [hunk 3]
```diff
@@ -730,7 +730,7 @@ describe("Bug", () => {
   })
 
   describe("in issue #10695", () => {
-    it("prevents showing MultiChoice's dropdown menu", async () => {
+    it.allowing(16)("prevents showing MultiChoice's dropdown menu", async () => {
```
Comment (bryevdv) [inline bokehjs/test/integration/regressions.ts:733]: That's funny, I thought this as allowed retries.

Comment (bryevdv) [reply bokehjs/test/integration/regressions.ts:733]: I guess it needs to be 22 for windows

Comment (bryevdv) [inline bokehjs/test/integration/regressions.ts:733]:
```suggestion
    it.allowing(22)("prevents showing MultiChoice's dropdown menu", async () => {
```

**Comment by bryevdv** [comment]
@mattpap if would be nice in cases where there is an `allowed` if the failure stated that value in addition to how many pixels actually differed.

**Comment by bryevdv** [comment]
@mattpap is the allowing value a strict upper bound? It's still failing with `allowing(22)`
```
✗ Bug
 └┬─ in issue #10507
  └── prevents changing MultiSelect.disabled property
images differ by 22px (0.09%)
```

File: bokehjs/test/integration/regressions.ts [hunk 4]
```diff
@@ -730,7 +730,7 @@ describe("Bug", () => {
   })
 
   describe("in issue #10695", () => {
-    it("prevents showing MultiChoice's dropdown menu", async () => {
+    it.allowing(22)("prevents showing MultiChoice's dropdown menu", async () => {
```
Comment (bryevdv) [inline bokehjs/test/integration/regressions.ts:733]:
```suggestion
    it.allowing(23)("prevents showing MultiChoice's dropdown menu", async () => {
```

File: bokehjs/test/integration/regressions.ts [hunk 5]
```diff
@@ -730,7 +730,7 @@ describe("Bug", () => {
   })
 
   describe("in issue #10695", () => {
-    it("prevents showing MultiChoice's dropdown menu", async () => {
+    it.allowing(23)("prevents showing MultiChoice's dropdown menu", async () => {
```
Comment (bryevdv) [inline bokehjs/test/integration/regressions.ts:733]: trying with 23

**Comment by bryevdv** [comment]
OK win test is still failing w/ 22 px difference, despite `allowing(23)` @mattpap if you have any other quick suggestions I am all ears otherwise I will kick this to a future release in order to get an rc built.

File: bokehjs/src/lib/models/callbacks/open_url.ts (hunk 2 above)
Comment (mattpap) [reply bokehjs/src/lib/models/callbacks/open_url.ts:32]: I guess I recall this from some old WinAPI nomenclature, but still this doesn't sound right to me. Maybe `navigate` (which is web standard call this)?

**Comment by mattpap** [comment]
@bryevdv, you've been changing the same test, but there is one more failing (the one on windows). I went ahead and resolved this, so tests should pass now.

File: bokehjs/src/lib/models/callbacks/open_url.ts (hunk 2 above)
Comment (mattpap) [reply bokehjs/src/lib/models/callbacks/open_url.ts]: I went ahead and renamed it to `navigate` when fixing tests.

**Comment by bryevdv** [comment]
>  you've been changing the same test

Is there any way to get line number information in the text failure output? I'd really appreciate an easy unambiguous pointer to the right test

**Comment by mattpap** [comment]
There are line numbers, but they point to generated JS files, not original TS source files. It's doable, but requires some work. Alternatively we could source map from TS files, but I don't do that, because browsers' devtools still don't handle TS well. I did some experimentation with this in PR #11017, and it was a nightmare to work with.

**Pull request merged**